src/testdir/test_z.vim

```diff
@@ -0,0 +1,69 @@
+" Test :z
+
+func Test_z()
+  call setline(1, range(1, 100))
+
+  let a=execute('20z3')
+  call assert_equal("\n20\n21\n22", a)
+  call assert_equal(22, line('.'))
+  " 'window' should be set to the {count} value.
+  call assert_equal(3, &window)
+
+  " If there is only one window, then twice the amount of 'scroll' is used.
+  set scroll=2
+  let a=execute('20z')
+  call assert_equal("\n20\n21\n22\n23", a)
+  call assert_equal(23, line('.'))
+
+  let a=execute('20z+3')
+  " FIXME: I would expect the same result as '20z3' but it
+  " gives "\n21\n22\n23" instead. Bug in Vim or in ":help :z"?
+  "call assert_equal("\n20\n21\n22", a)
+  "call assert_equal(22, line('.'))
```

Comment:
I did not fix the issue described in above comment. I discovered
it while writing the tests. I do not know whether it's a bug in :help :z 
or whether it's a bug in vim. Help page says:

`Specifying no mark at all is the same as "+"`

So I would expect commands ":20z3" and "20z+3" to be 
equivalent.  It looks like a bug in the :z  command, but it's
unrelated to the memory overflow fixed in this PR.

I had never used the  :z  command before.  It's probably
not used a lot.